icu4c/source/common/uvector.cpp

```diff
@@ -451,7 +486,8 @@ void UVector::sortedInsert(void* obj, UElementComparator *compare, UErrorCode& e
  * be sorted already.
  */
 void UVector::sortedInsert(int32_t obj, UElementComparator *compare, UErrorCode& ec) {
-    UElement e;
+    U_ASSERT(deleter == nullptr);
```

Comment:
Should the `sortedInsert` above maybe assert that there is a deleter?
`U_ASSERT(deleter != nullptr);`